content/en/tracing/trace_collection/custom_instrumentation/dotnet.md

```diff
@@ -180,7 +180,8 @@ using (var parentScope =
 
 The Datadog APM Tracer supports [B3][5] and [W3C][6] headers extraction and injection for distributed tracing. For more information, see the [setup documentation][7].
 
-In most cases, headers extraction and injection are transparent. Though, there are some known cases where your distributed trace can be disconnected. For instance, when reading messages from a distributed queue, some libraries may lose the span context. In that case, you can add a custom trace using the following code:
+In most cases, headers extraction and injection are transparent. In the case of distributed queues though, we don't know how the receiving end will be handled (consumed and processed, or consumed in a loop and stored for processing).
+In those cases (RabbitMQ, Kafka...), the tracer creates a short lived span when the message is consumed. We create this span mainly for tracking tags that would be relevant to you. Nevertheless, this span being immedialtely closed, any further work would not be attached to the current trace. If you need further work to be traced, you can do the following:
```

Comment:
```suggestion
In those cases (which include RabbitMQ and Kafka use), the tracer creates a short-lived span when the message is consumed, mainly for tracking tags that are relevant to you. But because this span is immediately closed, any further work is not attached to the current trace. If you need further work to be traced, you can add a custom trace using the following code:
```